src/append_only_data.rs

```diff
@@ -676,6 +698,28 @@ where
     P: Permissions + std::hash::Hash + Clone,
 {
     fn append(&mut self, entries: &[(Vec<u8>, Vec<u8>)]) -> Result<()> {
+        let existing: BTreeSet<Vec<u8>> = self
```

Comment:
This block seems to be the same as the previous one.
We can consider putting it into a function.